### DISCUSSION TITLE
`DbErr::RecordNotFound` instead of `DbErr::RecordNotUpdated` when the record doesn't exist

### DISCUSSION THREAD

**Discussion by daalfox** [SeaQL/sea-orm#1945] [Ideas]
I think when you try to call `update` on an `ActiveModel` that does not exist, it should return a `RecordNotFound` instead of `RecordNotUpdated`.

This makes it possible to try to update without knowing the record exists, when `update` returns `RecordNotUpdated` you don't really know that the record is missing or there is some other issue with update. This is specially helpful on rest apis when you receive the response and you know that you should return a 500 error or a 404 with a single database query.

If you agree I can make a PR

**Comment by tyt2y3** [comment]
Changing error variant is a breaking change, but it's worth considering.

However I am not sure whether we can actually distinguish between `RecordNotFound` and `RecordNotUpdated` in general.

Can you do some investigation on MySQL and Postgres to demonstrate the logic?

**Reply by daalfox** [reply]
Now that I'm thinking about it, it's not that beneficial, You need to query the whole table to find out the item is missing anyways
